packages/react-composites/src/composites/ChatComposite/FileDownloadCards.tsx

```diff
@@ -48,20 +52,34 @@ export const FileDownloadCards = (props: FileDownloadCards): JSX.Element => {
             fileName={truncatedFileName(file.name, truncateLength)}
             key={file.name}
             fileExtension={extension(file.name)}
-            actionIcon={<DownloadIconTrampoline />}
+            actionIcon={
+              showSpinner ? (
+                <Spinner size={SpinnerSize.medium} aria-live={'assertive'} />
+              ) : (
+                <ChatCompositeIcon iconName="Download" />
+              )
+            }
             actionHandler={() => {
-              props.downloadHandler && props.downloadHandler(userId, file);
+              if (props.downloadHandler) {
```

Comment:
This has become a long inline function, consider making this a standalone function inside the FileDownloadCards:
```
const fileDownloadHandler = useCallback(() => {
  // this big logic block
```

Reply:
Done